pca_ica/extract_cells.m

```diff
@@ -0,0 +1,326 @@
+function inv_quality = extract_cells(movie_source,pca_source,max_num)
+
+%   Cell extraction method that is based on an algorithm that essentially
+%   solves the sparse source separation problem to find cell filters.
+%   Traces are then found by using least squares.
+%
+%   Inputs:
+%
+%       movie_source: path to the movie matrix (height x width x frames)
+%
+%       pca_sources: path to the pca .mat file. It is assumed that there is
+%       a variable named pca_filters that has size [# of PCs x # of pixels]
+%       in the .mat file.
+%       
+%       max_num = Maximum estimated number of cells. Set to inf to get all
+%       that the algorithm can find in the movie.
+%
+%   Outputs:
+%
+%       Script saves the traces and filters in a .mat file.
+%
+%       inv_quality: Inverse quality metric that indicates how good each
+%       found potential is as far as the extraction algorithm is concerned.
+%       It is a [1 x # of extracted cells] array whose entries are
+%       normalized to [(1/# of pixels),1] range. This array is also plotted
+%       on the screen at the end of execution.
+%
+%   Hakan Inan (Aug 2015)
+%
+
+% Some parameters
+maxsize_chunk_GPU = 500;
+maxsize_chunk_CPU = 3000;
+corr_thresh = 0.1;
+filter_thresh = 0.25;
+
+max_num = min(max_num,1000); % Internal control for max_num
```

Comment:
Generally, if you are overwriting user-specified parameters, then it is common courtesy to indicate the override (e.g. by console output).